src/share/timing/perf_mod.F90

```diff
@@ -839,6 +841,95 @@ end subroutine t_stopf
+! Author: P. Worley (based on J. Rosinski GPTL routine)
+!-----------------------------------------------------------------------
+!---------------------------Input arguments-----------------------------
+!
+   ! performance timer event name
+   character(len=*), intent(in) :: event
+   ! walltime (seconds) associated with this start/stop pair
+   ! If not set, default is 0.0 . If < 0.0, set to 0.0 .
+   real(shr_kind_r8), intent(in), optional :: walltime
+   ! call count associated with this start/stop pair
+   ! If not set, default is 1. If < 0, set to 0.
+   integer, intent(in), optional :: callcount
+!
+!---------------------------Input/Output arguments----------------------
+!
+   ! GPTL event handle
+   integer,  optional :: handle
+!
+!---------------------------Local workspace-----------------------------
+!
+   integer  ierr                          ! GPTL error return
+   integer  str_length, i                 ! support for adding 
+                                          !  detail prefix
+   character(len=2) cdetail               ! char variable for detail
+   real(shr_kind_r8) wtime                ! walltime increment (seconds)
+   integer  callcnt                       ! call count increment
+!
+!-----------------------------------------------------------------------
+!
+   if (.not. timing_initialized) return
+   if (timing_disable_depth > 0) return
+
+   wtime = 0.0_shr_kind_r8
+   if ( present(walltime) ) then
+      if (walltime > 0.0) then
+         wtime = walltime
+      endif
+   endif
+
+   callcnt = 1
+   if ( present(callcount) ) then
+      if (callcount > 0) then
+         callcnt = callcount
+      else
+         callcnt = 0
+      endif
+   endif
+
+   if ((perf_add_detail) .AND. (cur_timing_detail < 100)) then
+
+      write(cdetail,'(i2.2)') cur_timing_detail
+      if (prefix_len > 0) then
+         str_length = min(SHR_KIND_CM-prefix_len-5,len_trim(event))
+         ierr = GPTLstartstop_vals( &
+            '"'//cdetail//"_"//event_prefix(1:prefix_len)// &
+            event(1:str_length)//'"', wtime, callcnt)
+      else
+         str_length = min(SHR_KIND_CM-5,len_trim(event))
+         ierr = GPTLstartstop_vals( &
+            '"'//cdetail//"_"//event(1:str_length)//'"', wtime, callcnt)
+      endif
+
+   else
+
+      if (prefix_len > 0) then
+         str_length = min(SHR_KIND_CM-prefix_len-2,len_trim(event))
+         ierr = GPTLstartstop_vals('"'//event_prefix(1:prefix_len)// &
+              event(1:str_length)//'"', wtime, callcnt)
+      else
+         str_length = min(SHR_KIND_CM-2,len_trim(event))
+         ierr = GPTLstartstop_vals('"'//trim(event)//'"', wtime, callcnt)
+      endif
+
+!pw   if ( present (handle) ) then
+!pw      ierr = GPTLstartstop_vals_handle(event, wtime, handle)
+!pw   else
+!pw      ierr = GPTLstartstop_vals(event, wtime)
+!pw   endif
+
+   endif
+
+   return
+   end subroutine t_startstop_valsf
+!
+!========================================================================
+!
```

Comment:
Significant changes to the timing library and output were made.  I have no specific objections as to the content of the changes, but find the process more than a little frustrating - there should be issues, discussion and pr's related to these changes.

Reply:
@jgfouca , yes, these are my changes. And there are more than just fortran changes (also C changes to gptl.c etc.) There were issues and PRs on the ACME side. Wanted to capture time spent prior to the call to initialize the timing library, since MPI_Init can cost as much as 20 minutes on Cori-KNL in some circumstances. Other tweaks were to: (a) more consistently add CPL: to cpl-related timers. Seemed an obvious clean-up, and I did not apply it everywhere, since it was delberately left off of a few timers and I do not know the history to these decisions, (b) force a checkpoint after the first simulated day, so that long initialization times can be captured even if a job runs out of wallclock time, (c) slightly widen the scope of the prefix application in component_init_cc, as this proved useful in performance debugging, and (d) do not add negative times to running totals (when the underlying timers is not monotonic).